src/components/inputs/InputField.tsx

```diff
@@ -0,0 +1,80 @@
+import { Box, TextField } from "@mui/material";
+import Error from "../error/Error";
+import { useFormContext } from "react-hook-form";
+import { ERRORS } from "../../assets/commons/text";
+
+export enum INPUT_FIELD_VARIANTS {
+  FILLED = "filled",
+  OUTLINED = "outlined",
+  STANDARD = "standard",
+}
+
+type InputProps = {
+  label: string;
+  type: string;
+  variant: INPUT_FIELD_VARIANTS;
+  required?: boolean;
+  dataName: string;
+  watchedInput?: string | undefined;
+};
+
+export default function InputField({
+  label,
+  type,
+  variant,
+  required = false,
+  dataName,
+  watchedInput = undefined,
```

Comment:
```suggestion
  watchedInput,
```

You don't have to have a default to this one, if it's not passed to the component it will be undefined by default)